ipatests/test_xmlrpc/test_automember_plugin.py

```diff
@@ -528,6 +689,171 @@ def test_rebuild_membership_hosts_group(self, automember_hostgroup, user1,
             command()
 
 
+class TestAutomemberAddConditionNegative(XMLRPC_test):
+
+    @pytest.mark.parametrize(
+        "automember_grp, grp", [
+            ("automember_group", "group1"),
+            ("automember_hostgroup", "hostgroup1")
+        ])
+    def test_create_inclusive_with_nonexistent_group(self, automember_grp,
+                                                     grp, request):
+        grp = request.getfixturevalue(grp)
+        automember_grp = request.getfixturevalue(automember_grp)
+        automember_grp.ensure_missing()
+        command = automember_grp.make_add_condition_command(
+            key=u'manager', type=automember_grp.membertype,
+            automemberinclusiveregex="eng[0-9]+.example.com")
+        with pytest.raises(errors.NotFound,
+                           match=r'Auto member rule: %s not found' % grp.cn):
+            command()
+
+    @pytest.mark.parametrize(
+        "automember_grp, grp", [
+            ("automember_group", "group1"),
+            ("automember_hostgroup", "hostgroup1")
+        ])
+    def test_create_exclusive_with_nonexistent_group(self, automember_grp,
+                                                     grp, request):
+        grp = request.getfixturevalue(grp)
+        automember_grp = request.getfixturevalue(automember_grp)
+        automember_grp.ensure_missing()
+        command = automember_grp.make_add_condition_command(
+            key=u'manager', type=automember_grp.membertype,
+            automemberexclusiveregex="qa[0-9]+.example.com")
+        with pytest.raises(errors.NotFound,
+                           match=r'Auto member rule: %s not found' % grp.cn):
+            command()
+
+    @pytest.mark.parametrize(
+        "automember_grp, grp", [
+            ("automember_group", "group1"),
+            ("automember_hostgroup", "hostgroup1")
+        ])
+    def test_create_inclusive_with_existent_group(self, automember_grp,
+                                                  grp, request):
+        grp = request.getfixturevalue(grp)
+        automember_grp = request.getfixturevalue(automember_grp)
+        grp.ensure_exists()
+        automember_grp.ensure_missing()
```

Comment:
The test name implies that we should have an existing group => automember_grp.ensure_exists()
No Exception is expected in this case.